packages/integrations/preact/src/server.ts

```diff
@@ -35,18 +37,50 @@ function check(Component, props, children) {
 	}
 }
 
-function renderToStaticMarkup(Component, props, { default: children, ...slotted }) {
-	const slots = {};
+function isSignal(x: any): x is SignalLike {
+	return x != null && typeof x === 'object' && typeof x.peek === 'function' && 'value' in x;
+}
+
+function renderToStaticMarkup(this: RendererContext, Component: any, props: Record<string, any>, { default: children, ...slotted }: Record<string, any>) {
+	const ctx = getContext(this.result);
+
+	const slots: Record<string, ReturnType<typeof h>> = {};
 	for (const [key, value] of Object.entries(slotted)) {
 		const name = slotName(key);
 		slots[name] = h(StaticHtml, { value, name });
 	}
 	// Note: create newProps to avoid mutating `props` before they are serialized
 	const newProps = { ...props, ...slots };
+
+	// Check for signals
+	const signals: Record<string, string> = {};
+	for(const [key, value] of Object.entries(props)) {
+		if(isSignal(value)) {
+			// Set the value to the current signal value
+			newProps[key] = value.peek();
```

Comment:
This seems wrong, actually. The component should be getting the actual signal, not the underlying value. We just want to grab the underlying value so that it is serialized correctly in the island.